src/v2/Utils/ExtractNodes.ts

```diff
@@ -0,0 +1,17 @@
+export function extractNodes<Node extends object, T = Node>(
```

Comment:
We should rename the file to match the function name 1:1 (will have to do a `git mv`, too)